Updates on cli reference documentation

vdemeester opened the pull request:
Made a pass on `cli` reference documentation 👼
- Update ps with `--last` flag
- Update commands with current output (including new ones :wink:)
- Make sure hugo does not detect the wrong language
- Update usage for `tag` command to be more coherent with the other ones

Should close #23749

/cc @thaJeztah @stevvooe (for `tag` 👼) @aaronlehmann @vieux (for swarm updates)

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

vdemeester commented on docs/reference/commandline/service_update.md:
```diff
@@ -12,26 +12,26 @@ parent = "smn_cli"
   -l, --label value                  Service labels (default [])
       --limit-cpu value              Limit CPUs (default 0.000)
       --limit-memory value           Limit Memory (default 0 B)
-      --mode string                  Service mode (replicated or global) (default "replicated")
```
This was documented but not in the actual code :sweat:

aaronlehmann replied:
The option was removed in #23697. Removing it from the docs is correct.

vdemeester commented on docs/reference/commandline/swarm_init.md:
````diff
@@ -50,6 +55,14 @@ $ docker swarm init --listen-addr 192.168.99.121:2377 --auto-accept worker
 Swarm initialized: current node (1m8cdsylxbf3lk8qriqt07hx1) is now a manager.
 ```
 
+### `--cert-expiry`
````
Pretty sure these description are either not useful at all or wrong :angel: 
@aaronlehmann @stevvooe I need your input :bow:

aaronlehmann commented on docs/reference/commandline/swarm_init.md:
```diff
@@ -50,6 +55,14 @@ $ docker swarm init --listen-addr 192.168.99.121:2377 --auto-accept worker
 
+### `--cert-expiry`
+
+This flag sets the validity period for node certificates.
```
Looks correct to me.

aaronlehmann commented on docs/reference/commandline/swarm_init.md:
```diff
@@ -50,6 +55,14 @@ $ docker swarm init --listen-addr 192.168.99.121:2377 --auto-accept worker
+
+### `--dispatcher-heartbeat`
+
+This flags sets the frequency with which manager will ask node for their health.
```
I think nodes announce their presence to a manager periodically rather than being asked by the manager. @stevvooe can confirm.

stevvooe replied:
Node's drive the heartbeat. This is the value that the dispatcher tells the nodes to use as a period.

vdemeester replied:
@stevvooe @aaronlehmann ok cool, thanks, I'll update then :wink:

thaJeztah commented on docs/reference/commandline/create.md:
```diff
@@ -12,95 +12,109 @@ parent = "smn_cli"
+      --oom-kill-disable            Disable OOM Killer
+      --oom-score-adj int           Tune host's OOM preferences (-1000 to 1000)
+      --pid string                  PID namespace to use
+      --pids-limit int              Tune container pids limit (set -1 for unlimited)
```
Was the previous `kernel >= 4.3` manually added to the docs, or part of the usage output? Wondering if we need it somewhere

vdemeester replied:
If it's not there, manuallh added :sweat:. I should either add it back manually or add it in the flag usage, wdyt ?

thaJeztah replied:
not really sure, we don't mention it on other options, do we output a warning on `docker info` if it's not supported?

vdemeester replied:
no idea :sweat_smile:

thaJeztah replied:
for now, we could add it back in the docs only

thaJeztah commented on docs/reference/commandline/create.md:
```diff
@@ -12,95 +12,109 @@ parent = "smn_cli"
+      --restart string              Restart policy to apply when a container exits (default "no")
+      --runtime string              Runtime to use for this container
+      --security-opt value          Security Options (default [])
+      --shm-size string             Size of /dev/shm, default value is 64MB.
```
Hm, wasnt there a change for the size recently, or was that for tmpfs? Need to check

vdemeester replied:
I think it was tmpfs :angel:

thaJeztah commented on docs/reference/commandline/run.md:
```diff
@@ -10,109 +10,119 @@ parent = "smn_cli"
+  -P, --publish-all                 Publish all exposed ports to random ports
       --read-only                   Mount the container's root filesystem as read only
-      --restart="no"                Restart policy (no, on-failure[:max-retry], always, unless-stopped)
+      --restart string              Restart policy to apply when a container exits (default "no")
```
Perhaps we should keep the available options here (ideally in the help output as well, but there's more places where we have that issue)

thaJeztah replied:
oh! I think we need it on `docker create` as well then (sorry!)

vdemeester replied:
@thaJeztah you are right

vdemeester replied:
… and done :wink:

thaJeztah commented:
Thanks! Some minor notes, but LGTM once addressed

thaJeztah commented:
Also we should have some tooling again to update these automatically (we once had)

thaJeztah commented:
LGTM, thanks!

ping @sfsmithcha PTAL

sfsmithcha commented on docs/reference/commandline/node_rm.md:
````diff
@@ -11,17 +11,19 @@ parent = "smn_cli"
+```
 
-Removes nodes that are specified. 
+Removes nodes that are specified.
````
Possible to mention, used to "reject" pending nodes. https://github.com/docker/docker/issues/24375

Also, could be tighter suggest:

Removes specified nodes from a swarm. Rejects nodes with `Pending` membership from the swarm.

sfsmithcha commented:
Other than minor comments, LGTM.

vdemeester committed:
Updates on cli reference documentation

- Update ps with `--last` flag
- Update commands with current output
- Make sure hugo does not detect the wrong language
- Update usage for `tag` command to be more coherent with the other ones

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
@thaJeztah @sfsmithcha Updated :wink:

sfsmithcha commented:
@vdemeester LGTM. I actually didn't save my other comments inline which were:

* remove code tics `--some-flag` in the h3s. they mess up the format. @thaJeztah and I had discussed this. There is also the question of would you also need to remove the `--` at the beginning in this case.

* also swarm is normally lowercase: "a swarm" not a proper noun. The could be a separate issue: clarify usage of "swarm" in Engine docs.

Neither of these are critical for this PR IMO.

thaJeztah commented:
Thanks @sfsmithcha! Let's do those in a follow up all green and WindowsTP5 is in "pending" (but windowsRS1 is green) so I'm merging this

stevvooe commented:
❤️
